server/src/main/java/io/crate/execution/dsl/phases/PKLookupPhase.java

```diff
@@ -47,14 +49,14 @@ public final class PKLookupPhase extends AbstractProjectionsPhase implements Col
 
     private final List<ColumnIdent> partitionedByColumns;
     private final List<Symbol> toCollect;
-    private final Map<String, Map<ShardId, List<PKAndVersion>>> idsByShardByNode;
+    private final Map<String, Map<ShardId, Set<PKAndVersion>>> idsByShardByNode;
```

Comment:
Given that it is order sensitive I think all the `Set` parts should be replaced with `SequencedSet`

Reply:
done, thx!